test/miner_ct_utils.erl

```diff
@@ -857,6 +858,39 @@ init_base_dir_config(Mod, TestCase, Config)->
         | Config
     ].
 
+%%--------------------------------------------------------------------
+%% @doc
+%% Wait for a particular txn type to occur on miners
+%% @end
+%%-------------------------------------------------------------------
+wait_for_txn(Miners, TxnType) ->
+    wait_for_txn(Miners, TxnType, timer:seconds(30)).
+
+wait_for_txn(Miners, TxnType, Timeout)->
+    ?assertAsync(begin
+                     Result = lists:all(
+                                fun(Miner) ->
+                                        C = ct_rpc:call(Miner, blockchain_worker, blockchain, [], Timeout),
+                                        H = ct_rpc:call(Miner, blockchain, height, [C], Timeout),
+                                        Blocks = ct_rpc:call(Miner, blockchain, blocks, [C], Timeout),
+
+                                        Res = lists:filter(fun({_Hash, Block}) ->
+                                                                   BH = blockchain_block:height(Block),
+                                                                   Txns = blockchain_block:transactions(Block),
+                                                                   ToFind = lists:filter(fun(T) ->
+                                                                                                 blockchain_txn:type(T) == TxnType
```

Comment:
Make this a predicate fun that's passed in instead of TxnType

Reply:
Good idea

Reply:
Check: fa0efd0